src/kw/kw.cpp

```diff
@@ -146,23 +188,48 @@ string KanoWorld::get_refresh_header(string token)
 
 
 /**
- *   libcurl callback function to collect response from the server
- *
+ *   libcurl callback function, allows to collect the data response from the server,
+ *   which in this case is a json object.
  */
-size_t KanoWorld::write_function(void *ptr, size_t size, size_t nmemb, void *stream)
+size_t KanoWorld::write_function(void *ptr, size_t size, size_t nmemb, void *user_data)
 {
-    // TODO: Will be needed to parse the json response. How to collect class this pointer?
-    //printf(">>> RESPONSE: %s\n", ptr);
+    // Save the server's response in a class string
+    KanoWorld *pkw = (KanoWorld *) user_data;
```

Comment:
This is horrible. What is this guy doing? If we need to do it, can we make the parameters smart pointers (`std::shared_ptr`, etc). If not, do something else. Either way, from the function name I have no idea what it is doing.

Reply:
This is one of the explanations on how this works:

 * https://stackoverflow.com/questions/7974371/libcurl-callbacks-w-c-class-members

Good point on using smart pointers.

Reply:
As discussed, we'll leave it as it is and migrate later to your solution